diff --git a/kernel/supercall/dispatch.c.1 b/kernel/supercall/dispatch.c
index 9232d810..5466114e 100644
--- a/kernel/supercall/dispatch.c.1	2026-10-08 23:20:27.696409400 -0700
+++ b/kernel/supercall/dispatch.c	2026-10-08 23:30:13.201328800 -0700
@@ -76,17 +76,14 @@
 
 #ifdef MODULE
     cmd.flags |= KSU_GET_INFO_FLAG_LKM;
-    if (ksu_bundled) {
-        cmd.flags |= KSU_GET_INFO_FLAG_BUNDLED;
-    }
 #endif
 
     if (is_manager()) {
         cmd.flags |= KSU_GET_INFO_FLAG_MANAGER;
     }
-    if (ksu_late_loaded) {
-        cmd.flags |= KSU_GET_INFO_FLAG_LATE_LOAD;
-    }
+#ifdef EXPECTED_SIZE2
+    cmd.flags |= KSU_GET_INFO_FLAG_PR_BUILD;
+#endif
     cmd.features = KSU_FEATURE_MAX;
 
     if (copy_to_user(arg, &cmd, sizeof(cmd))) {
@@ -707,11 +704,7 @@
 {
     struct ksu_get_hook_mode_cmd cmd = {0};
 
-#ifdef CONFIG_HAVE_SYSCALL_TRACEPOINTS
-    strscpy(cmd.mode, "Tracepoint", sizeof(cmd.mode));
-#else
-    strscpy(cmd.mode, "Kprobes", sizeof(cmd.mode));
-#endif
+    strscpy(cmd.mode, "Inline (SuSFS)", sizeof(cmd.mode));
 
     if (copy_to_user(arg, &cmd, sizeof(cmd))) {
         pr_err("get_hook_mode: copy_to_user failed\n");
