diff --git a/kernel/feature/selinux_hide.c b/kernel/feature/selinux_hide.c
index 033cab8dc..b1d349b26 100644
--- a/kernel/feature/selinux_hide.c
+++ b/kernel/feature/selinux_hide.c
@@ -87,7 +87,7 @@
         return orig_context_write(file, buf, size);
     }
     char *canon = NULL;
-    u32 sid, len;
+    u32 sid, len, tmp;
     ssize_t length;
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)
@@ -95,8 +95,12 @@
     if (length)
         goto out;
     length = security_context_to_sid_with_policy(backup_sepolicy, buf, size, &sid, SECSID_NULL, GFP_KERNEL);
-    if (length)
+    if (length) {
         goto out;
+    } else {
+        // sync to global sidtab
+        security_context_to_sid(buf, size, &tmp, GFP_KERNEL);
+    }
 
     length = security_sid_to_context_with_policy(backup_sepolicy, sid, &canon, &len);
     if (length)
@@ -116,8 +120,12 @@
         goto out;
 
     length = security_context_to_sid(&fake_state, buf, size, &sid, GFP_KERNEL);
-    if (length)
+    if (length) {
         goto out;
+    } else {
+        // sync to global sidtab
+        security_context_to_sid(&selinux_state, buf, size, &tmp, GFP_KERNEL);
+    }
 
     length = security_sid_to_context(&fake_state, sid, &canon, &len);
     if (length)
@@ -138,7 +146,7 @@
         return orig_access_write(file, buf, size);
     }
     char *scon = NULL, *tcon = NULL;
-    u32 ssid, tsid;
+    u32 ssid, tsid, sconlen, tconlen, tmp;
     u16 tclass;
     struct av_decision avd;
     ssize_t length;
@@ -166,24 +174,43 @@
     if (sscanf(buf, "%s %s %hu", scon, tcon, &tclass) != 3)
         goto out;
 
+    sconlen = strlen(scon);
+    tconlen = strlen(tcon);
+
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)
-    length = security_context_to_sid_with_policy(backup_sepolicy, scon, strlen(scon), &ssid, SECSID_NULL, GFP_KERNEL);
-    if (length)
+    length = security_context_to_sid_with_policy(backup_sepolicy, scon, sconlen, &ssid, SECSID_NULL, GFP_KERNEL);
+    if (length) {
         goto out;
+    } else {
+        // sync to global sidtab
+        security_context_to_sid(scon, sconlen, &tmp, GFP_KERNEL);
+    }
 
-    length = security_context_to_sid_with_policy(backup_sepolicy, tcon, strlen(tcon), &tsid, SECSID_NULL, GFP_KERNEL);
-    if (length)
+    length = security_context_to_sid_with_policy(backup_sepolicy, tcon, tconlen, &tsid, SECSID_NULL, GFP_KERNEL);
+    if (length) {
         goto out;
+    } else {
+        // sync to global sidtab
+        security_context_to_sid(tcon, tconlen, &tmp, GFP_KERNEL);
+    }
 
     security_compute_av_user_with_policy(backup_sepolicy, ssid, tsid, tclass, &avd);
 #else
-    length = security_context_str_to_sid(&fake_state, scon, &ssid, GFP_KERNEL);
-    if (length)
+    length = security_context_to_sid(&fake_state, scon, sconlen, &ssid, GFP_KERNEL);
+    if (length) {
         goto out;
+    } else {
+        // sync to global sidtab
+        security_context_to_sid(&selinux_state, scon, sconlen, &tmp, GFP_KERNEL);
+    }
 
-    length = security_context_str_to_sid(&fake_state, tcon, &tsid, GFP_KERNEL);
-    if (length)
+    length = security_context_to_sid(&fake_state, tcon, tconlen, &tsid, GFP_KERNEL);
+    if (length) {
         goto out;
+    } else {
+        // sync to global sidtab
+        security_context_to_sid(&selinux_state, tcon, tconlen, &tmp, GFP_KERNEL);
+    }
 
     security_compute_av_user(&fake_state, ssid, tsid, tclass, &avd);
 #endif
@@ -204,7 +231,7 @@
 typedef int (*setprocattr_fn)(const char *name, void *value, size_t size);
 static int __nocfi my_setprocattr(const char *name, void *value, size_t size)
 {
-    int error;
+    int error, perm_error;
     u32 mysid, sid;
     char *str = value;
     if (likely(current_uid().val < 10000)) {
@@ -214,17 +241,6 @@
     if (strcmp(name, "current")) {
         goto call_orig;
     }
-    mysid = current_sid();
-
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)
-    error = avc_has_perm(mysid, mysid, SECCLASS_PROCESS, PROCESS__SETCURRENT, NULL);
-#else
-    error = avc_has_perm(&selinux_state, mysid, mysid, SECCLASS_PROCESS, PROCESS__SETCURRENT, NULL);
-#endif
-    if (error) {
-        return error;
-    }
-
     if (size && str[0] && str[0] != '\n') {
         if (str[size - 1] == '\n') {
             str[size - 1] = 0;
@@ -236,7 +252,13 @@
         error = security_context_to_sid(&fake_state, str, size, &sid, GFP_KERNEL);
 #endif
         if (error) {
-            return error;
+            mysid = current_sid();
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)
+            perm_error = avc_has_perm(mysid, mysid, SECCLASS_PROCESS, PROCESS__SETCURRENT, NULL);
+#else
+            perm_error = avc_has_perm(&selinux_state, mysid, mysid, SECCLASS_PROCESS, PROCESS__SETCURRENT, NULL);
+#endif
+            return perm_error ?: error;
         }
     }
 
 
